### PR TITLE
Provide more info when cookbook metadata is not found

### PR DESCRIPTION
**Pull request by jdmundrawala**
cc @danielsdeleo @sethvargo @opscode/client-engineers

### CODE REVIEW

File: lib/chef/exceptions.rb [hunk 1]
```diff
@@ -152,8 +152,12 @@ class IllegalVersionConstraint < NotImplementedError; end
 
     class MetadataNotValid < StandardError; end
     class MetadataNotFound < StandardError
-      def initialize
-        super "No metadata.rb or metadata.json!"
+      attr_reader :install_path
+      attr_reader :cookbook_name
+      def initialize(install_path, cookbook_name)
+        @install_path = install_path
+        @cookbook_name = cookbook_name
```
Comment (sethvargo) [inline lib/chef/exceptions.rb]: I don't think we need to convert these to instance variables - just pass in the method args directly probably. Thoughts?

Comment (jdmundrawala) [inline lib/chef/exceptions.rb]: Main reason was that it made it easy to test.

**Comment by danielsdeleo** [comment]
:+1: :100:

**Comment by lamont-granquist** [comment]
:+1:

**Comment by jdmundrawala** [comment]
Merged into 12-stable:
https://github.com/opscode/chef/commit/b1dedaa6e1c3b8f70c16895c7c221286409f9564

**Comment by jdmundrawala** [comment]
Merged into 11-stable:
https://github.com/opscode/chef/commit/7a6f7d5fa16a906ef2b0e6582c9458b3369573fb

**Pull request merged**